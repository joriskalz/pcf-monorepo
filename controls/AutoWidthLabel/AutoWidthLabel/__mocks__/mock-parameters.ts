/* istanbul ignore file */

import { IInputs } from "../generated/ManifestTypes";
import {
  MockDecimalNumberProperty,
  MockEnumProperty,
  MockStringProperty,
  MockWholeNumberProperty,
} from "./mock-context";

export function getMockParameters(): IInputs {
  return {
    Text: new MockStringProperty(),
    PaddingLeft: new MockWholeNumberProperty(),
    PaddingRight: new MockWholeNumberProperty(),
    PaddingTop: new MockWholeNumberProperty(),
    PaddingBottom: new MockWholeNumberProperty(),
    FontName: new MockStringProperty(),
    FontSize: new MockDecimalNumberProperty(),
    FontSizeUnits: new MockEnumProperty("1"),
    FontColor: new MockStringProperty(),
    FontWeight: new MockStringProperty(),
    DisabledFontColor: new MockStringProperty(),
    DisabledFontWeight: new MockStringProperty(),
    FocusFontColor: new MockStringProperty(),
    FocusFontWeight: new MockStringProperty(),
    HoverFontColor: new MockStringProperty(),
    HoverFontWeight: new MockStringProperty(),
    FillColor: new MockStringProperty(),
    DisabledFillColor: new MockStringProperty(),
    FocusFillColor: new MockStringProperty(),
    HoverFillColor: new MockStringProperty(),
    //DisabledHoverFillColor: new MockStringProperty(),
    BorderColor: new MockStringProperty(),
    BorderThickness: new MockWholeNumberProperty(),
    BorderRadius: new MockWholeNumberProperty(),
    DisabledBorderColor: new MockStringProperty(),
    FocusBorderColor: new MockStringProperty(),
    FocusBorderThickness: new MockWholeNumberProperty(),
    HoverBorderColor: new MockStringProperty(),
    HoverBorderThickness: new MockWholeNumberProperty(),
    //RenderBorderStyle: new MockEnumProperty<"0" | "1">(),
  };
}
