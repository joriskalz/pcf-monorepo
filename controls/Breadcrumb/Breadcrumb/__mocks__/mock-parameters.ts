/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
        MaxDisplayedItems: new MockWholeNumberProperty(),
        OverflowIndex: new MockWholeNumberProperty(),
    };
}
