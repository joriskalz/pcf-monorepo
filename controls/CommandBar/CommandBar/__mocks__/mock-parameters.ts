/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
