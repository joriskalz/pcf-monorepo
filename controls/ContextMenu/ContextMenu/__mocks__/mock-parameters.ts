/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        TextAlignment: new MockEnumProperty(),
        items: new MockDataSet([]),
        Chevron: new MockTwoOptionsProperty(),
        FillColor: new MockStringProperty(),
        BorderColor: new MockStringProperty(),
        BorderRadius: new MockWholeNumberProperty(),
        FontColor: new MockStringProperty(),
        FontSize: new MockWholeNumberProperty(),
        HoverBorderColor: new MockStringProperty(),
        HoverFillColor: new MockStringProperty(),
        HoverFontColor: new MockStringProperty(),
        HoverIconColor: new MockStringProperty(),
        IconColor: new MockStringProperty(),
        IconSize: new MockWholeNumberProperty(),
    };
}
