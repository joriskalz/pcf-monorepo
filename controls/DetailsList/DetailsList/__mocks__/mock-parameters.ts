/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Compact: new MockTwoOptionsProperty(),
        CurrentSortColumn: new MockStringProperty(),
        CurrentSortDirection: new MockEnumProperty('0'),
        PageSize: new MockWholeNumberProperty(),
        RaiseOnRowSelectionChangeEvent: new MockTwoOptionsProperty(),
        SelectRowsOnFocus: new MockTwoOptionsProperty(),
        SelectionType: new MockEnumProperty('0'),
        AlternateRowColor: new MockStringProperty(),
        HeaderVisible: new MockTwoOptionsProperty(false),
        SelectionAlwaysVisible: new MockTwoOptionsProperty(false),
        records: new MockDataSet([]),
        columns: new MockDataSet([]),
        LargeDatasetPaging: new MockTwoOptionsProperty(),
    };
}
