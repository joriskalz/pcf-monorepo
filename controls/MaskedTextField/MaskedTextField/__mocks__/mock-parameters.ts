/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockTwoOptionsProperty } from './mock-context';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Value: new MockStringProperty(),
        Prefix: new MockStringProperty(),
        Suffix: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        ErrorMessage: new MockStringProperty(),
        Mask: new MockStringProperty(),
        MaskFormat: new MockStringProperty(),
        DelayOutput: new MockTwoOptionsProperty(),
    };
}
