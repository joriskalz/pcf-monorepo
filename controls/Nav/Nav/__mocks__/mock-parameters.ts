/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockTwoOptionsProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        CollapseByDefault: new MockTwoOptionsProperty(),
        InputEvent: new MockStringProperty(),
        SelectedKey: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
