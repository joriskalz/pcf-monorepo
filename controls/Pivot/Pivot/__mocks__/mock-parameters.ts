/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        RenderSize: new MockEnumProperty(),
        RenderType: new MockEnumProperty(),
        SelectedKey: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
