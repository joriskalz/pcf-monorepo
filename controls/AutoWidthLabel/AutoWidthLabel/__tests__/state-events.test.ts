import { AutoWidthLabel } from "..";
import { IInputs } from "../generated/ManifestTypes";
import { MockContext, MockState } from "../__mocks__/mock-context";
import { getMockParameters } from "../__mocks__/mock-parameters";

describe("ResizeableTextArea element state events", () => {
  beforeEach(() => jest.clearAllMocks());

  it("updates states on hover and focus", () => {
    const { component, context, notifyOutputChanged, state, container } = createComponent();
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (context as any).accessibility.assignedTabIndex = 10;
    getClassicStyles(context);

    component.init(context, notifyOutputChanged, state, container);
    expect(container).toMatchSnapshot();

    // Set mouse over
    const element = container.getElementsByClassName("cat-label-text")[0];
    element.dispatchEvent(new MouseEvent("mouseenter"));
    expect(container).toMatchSnapshot("hover");

    // Set mouse out
    element.dispatchEvent(new MouseEvent("mouseleave"));
    expect(container).toMatchSnapshot("hover:leave");

    // Set focus
    element.dispatchEvent(new MouseEvent("focus"));
    expect(container).toMatchSnapshot("focus");

    // Set lose focus
    element.dispatchEvent(new MouseEvent("blur"));
    expect(container).toMatchSnapshot("blur");
  });

  it("sets disabled styles", () => {
    const { context, container, component, notifyOutputChanged, state } = createComponent();
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    context.mode.isControlDisabled = true;
    getClassicStyles(context);
    component.init(context, notifyOutputChanged, state, container);
    expect(container).toMatchSnapshot("disabled");
  });
});

function getClassicStyles(context: MockContext<IInputs>) {
  const { parameters } = context;
  parameters.PaddingLeft.raw = 12;
  parameters.PaddingRight.raw = 5;
  parameters.PaddingTop.raw = 5;
  parameters.PaddingBottom.raw = 5;
  parameters.FontSize.raw = 17.5;
  parameters.FontColor.raw = "rgb(0,0,0)";
  parameters.FontWeight.raw = "400";
  parameters.DisabledFontColor.raw = "rgb(166,166,166)";
  parameters.FillColor.raw = "#ffffff";
  parameters.DisabledFillColor.raw = "rgb(244,244,244)";
  parameters.HoverFillColor.raw = "#BACAE2";
  parameters.BorderColor.raw = "#00126B";
  parameters.BorderThickness.raw = 2;
  parameters.BorderRadius.raw = 5;
  parameters.DisabledBorderColor.raw = "rgb(166,166,166)";
  parameters.FocusBorderThickness.raw = 4;
}

function createComponent() {
  const component = new AutoWidthLabel();
  const notifyOutputChanged = jest.fn();
  const context = new MockContext<IInputs>(getMockParameters());
  const state = new MockState();
  const container = document.createElement("div");
  document.body.appendChild(container);
  return { component, context, notifyOutputChanged, state, container };
}
