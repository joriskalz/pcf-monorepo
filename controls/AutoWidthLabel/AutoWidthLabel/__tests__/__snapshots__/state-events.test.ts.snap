// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResizeableTextArea element state events sets disabled styles: disabled 1`] = `
<div>
  <div
    class="cat-label-text-container"
  >
    <div
      class="cat-label-text"
      style="font-size: 17.5px; padding: 5px 5px 5px 12px; border-radius: 5px; color: rgb(166, 166, 166); font-weight: 400; border-width: 2px; border-color: rgb(166,166,166); background: rgb(244, 244, 244);"
    />
  </div>
</div>
`;

exports[`ResizeableTextArea element state events updates states on hover and focus 1`] = `
<div>
  <div
    class="cat-label-text-container"
  >
    <div
      class="cat-label-text"
      style="font-size: 17.5px; padding: 5px 5px 5px 12px; border-radius: 5px; color: rgb(0, 0, 0); font-weight: 400; border-width: 2px; border-color: #00126b; background: rgb(255, 255, 255);"
      tabindex="10"
    />
  </div>
</div>
`;

exports[`ResizeableTextArea element state events updates states on hover and focus: blur 1`] = `
<div>
  <div
    class="cat-label-text-container"
  >
    <div
      class="cat-label-text"
      style="font-size: 17.5px; padding: 5px 5px 5px 12px; border-radius: 5px; color: rgb(0, 0, 0); font-weight: 400; border-width: 2px; border-color: #00126b; background: rgb(255, 255, 255);"
      tabindex="10"
    />
  </div>
</div>
`;

exports[`ResizeableTextArea element state events updates states on hover and focus: focus 1`] = `
<div>
  <div
    class="cat-label-text-container"
  >
    <div
      class="cat-label-text"
      style="font-size: 17.5px; padding: 5px 5px 5px 12px; border-radius: 5px; color: rgb(0, 0, 0); font-weight: 400; border-width: 4px; border-color: #00126b; background: rgb(255, 255, 255);"
      tabindex="10"
    />
  </div>
</div>
`;

exports[`ResizeableTextArea element state events updates states on hover and focus: hover 1`] = `
<div>
  <div
    class="cat-label-text-container"
  >
    <div
      class="cat-label-text"
      style="font-size: 17.5px; padding: 5px 5px 5px 12px; border-radius: 5px; color: rgb(0, 0, 0); font-weight: 400; border-width: 2px; border-color: #00126b; background: rgb(186, 202, 226);"
      tabindex="10"
    />
  </div>
</div>
`;

exports[`ResizeableTextArea element state events updates states on hover and focus: hover:leave 1`] = `
<div>
  <div
    class="cat-label-text-container"
  >
    <div
      class="cat-label-text"
      style="font-size: 17.5px; padding: 5px 5px 5px 12px; border-radius: 5px; color: rgb(0, 0, 0); font-weight: 400; border-width: 2px; border-color: #00126b; background: rgb(255, 255, 255);"
      tabindex="10"
    />
  </div>
</div>
`;
